pybaseball/team_batting_against.py

```diff
@@ -27,23 +27,27 @@ def get_team_batting_against(year: int) -> pd.DataFrame:
 def team_batting_against(season: Optional[int] = None) -> pd.DataFrame:
     if season is None:
         season = most_recent_season()
+    elif season < 1915 or season > most_recent_season():
+        raise ValueError("Team Batting Stats available between 1915 and current season")
+
     team_batting_against = get_team_batting_against(season)
     team_batting_against = pd.concat(team_batting_against)
     team_batting_against = postprocess(team_batting_against)
     return team_batting_against
 
 
 def postprocess(team_batting_against: pd.DataFrame) -> pd.DataFrame:
-    # drop usually empty PA for which data is not known column
-    team_batting_against = team_batting_against.drop('PAu', 1)
+    # skip duplicative header in penultimate row, keep league average and total
+    team_batting_against = team_batting_against.drop(len(team_batting_against) - 2)
+    team_batting_against.reset_index(drop=True, inplace=True)
 
-    # skip league average and duplicative header
-    team_batting_against = team_batting_against.iloc[:30]
+    # fix Totals name in Tm column
+    team_batting_against.iat[len(team_batting_against) - 1, 0] = 'Total'
 
     # fix numeric conversion now that string values are removed
     postprocessing.convert_numeric(team_batting_against, postprocessing.columns_except(team_batting_against, ['Tm']))
-    int_cols = ["G", "PA", "AB", "R", "H", "2B", "3B", "HR", "SB",
-                "CS", "BB", "SO", "TB", "GDP", "HBP", "SH", "SF", "IBB", "ROE"]
-    team_batting_against.loc[:, int_cols] = team_batting_against.loc[:, int_cols].astype(int)
+    # int_cols = ["PAu", "G", "PA", "AB", "R", "H", "2B", "3B", "HR", "SB",
+    #             "CS", "BB", "SO", "TB", "GDP", "HBP", "SH", "SF", "IBB", "ROE"]
+    # team_batting_against[int_cols] = team_batting_against.loc[:, int_cols].astype(int)
```

Comment:
Is there a reason to keep this in as a comment?

Reply:
Yeah good catch, I had meant to do this conversion that was NA safe. Replaced this with conversion to `Int64`. Do we have a preferred pattern for addressing NA values in result dataframes, or do we leave that for the user to determine?